### PR TITLE
Make updates to modals to be more generic

### PR DESCRIPTION
**Pull request by jondauz**
Added new features to Modal component to be more generic.

### CODE REVIEW

File: src/css/components/Modal.scss [hunk 1]
```diff
@@ -34,14 +34,13 @@
...
-}
-
-.Modal-header {
-  text-align: center;
-}
+.Modal-close {
+  position: absolute;
+  top: 0.5rem;
+  right: 0.5rem;
+}
```
Comment (BillyFigueroa) [inline src/css/components/Modal.scss:46]: this needs a new line according to github

File: src/js/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -1,43 +1,57 @@
...
+      }
+    }
+  },[]);
+
   return (
     <div className={`Modal ${show ? 'Modal--open' : 'Modal--closed'}`}>
-      <div className='Modal-content'>
-        <div className='Modal-header'>
-          <h2 style={{ marginTop: '20px' }}>{heading}</h2>
+      <div className={`Modal-content u-posRelative`} style={{ width: '50%', ...(style || {})}}>
```
Comment (BillyFigueroa) [inline src/js/components/Modal/Modal.tsx:42]: should `className` just be a string if we are not dynamically generating the string?

Comment (jondauz) [reply src/js/components/Modal/Modal.tsx]: I absolutely agree. Latest check-in  should fix this.

File: src/js/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -1,43 +1,57 @@
...
-      <div className='Modal-content'>
-        <div className='Modal-header'>
-          <h2 style={{ marginTop: '20px' }}>{heading}</h2>
+      <div className='Modal-content u-posRelative' style={{ width: '50%', ...(style || {})}}>
+        <div className='Modal-header u-flex u-flexJustifyBetween'>
+          <div className="u-sizeFill">
+            <h2>{heading}</h2>
+          </div>
+          { showClose &&
+            <div className="Modal-close">
```
Comment (BillyFigueroa) [inline src/js/components/Modal/Modal.tsx:48]: Let's stick to using 1 type of quote. Single quotes are usually recommended. Sorry to nitpick here there are other cases up above where we are using double quotes again

File: src/js/components/Modal/Modal.stories.tsx [hunk 4]
```diff
@@ -13,13 +13,13 @@ stories.add('Default', () => {
...
         label='Test Modal'
-        onClick={ () => setSending(true) }
+        onClick={() => setSending(true)}
       />
 
       <Modal
-        heading='Sending'
-        show={ sending }
-        handleCTA={() => setSending(false)}
+        heading={<div className="u-textCenter">Sending</div>}
```
Comment (BillyFigueroa) [inline src/js/components/Modal/Modal.stories.tsx:20]: please use single quotes

File: src/js/components/Modal/Modal.stories.tsx [hunk 5]
```diff
@@ -31,7 +31,37 @@ stories.add('Default', () => {
...
+          setTimeout(() => {
+            setSending(false);
+          }, 5000);
+
+          setSending(true);
+        }}
+      />
+
+      <Modal
+        heading={<div className="u-textCenter">Self closing modal</div>}
```
Comment (BillyFigueroa) [inline src/js/components/Modal/Modal.stories.tsx:51]: Please use single quotes

Comment (bradcypert) [reply src/js/components/Modal/Modal.stories.tsx:51]: Created https://github.com/teamsnap/teamsnap-ui/issues/691 to help prevent minor syntax issues like this in the future. I'd love for you to take that and run with it if you'd like to, Billy!

Comment (jondauz) [reply src/js/components/Modal/Modal.stories.tsx]: Double quotes changed to single in the latest check-in

**Review by BillyFigueroa — APPROVED** [review]

**Pull request merged**